### PR TITLE
[fix](Nereids) fix fe fold constant with date time out of range

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

When running sql:
select date_add('9999-12-31 23:59:59', interval 5 second);
it should throw exception but not return null value

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [x] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [x] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by LiBinfeng-01** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/DateTimeLiteral.java [hunk 1]
```diff
@@ -395,9 +396,10 @@ public LocalDateTime toJavaDateType() {
     }
 
     public static Expression fromJavaDateType(LocalDateTime dateTime) {
-        return isDateOutOfRange(dateTime)
-                ? new NullLiteral(DateTimeType.INSTANCE)
-                : new DateTimeLiteral(dateTime.getYear(), dateTime.getMonthValue(), dateTime.getDayOfMonth(),
+        if (isDateOutOfRange(dateTime)) {
+            throw new NotSupportedException("date out of range");
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/DateTimeLiteral.java:400]: should print the detail message for user to debug

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Comment by englefly** [comment]
return null 才是正确的返回方式
类比 除以0的情况

**Review by zclllyybb — APPROVED** [review]

**Comment by zclllyybb** [comment]
> return null 才是正确的返回方式 类比 除以0的情况

可能恰恰相反啊。。有些return null的，我们反而要逐渐改到报错去

**Pull request merged**